ui/app/services/flags.ts

```diff
@@ -24,22 +25,37 @@ export default class flagsService extends Service {
   @service declare readonly version: VersionService;
   @service declare readonly store: StoreService;
 
-  @tracked flags: string[] = [];
   @tracked activatedFlags: string[] = [];
+  @tracked featureFlags: string[] = [];
 
-  setFeatureFlags(flags: string[]) {
-    this.flags = flags;
+  get isHvdManaged(): boolean {
+    return this.featureFlags?.includes(FLAGS.vaultCloudNamespace);
   }
 
-  get managedNamespaceRoot() {
-    if (this.flags && this.flags.includes(FLAGS.vaultCloudNamespace)) {
-      return 'admin';
+  get hvdManagedNamespaceRoot(): string | null {
+    return this.isHvdManaged ? 'admin' : null;
+  }
+
+  getFeatureFlags = keepLatestTask(async () => {
+    try {
+      const result = await fetch('/v1/sys/internal/ui/feature-flags', {
+        method: 'GET',
+      });
+
+      if (result.status === 200) {
```

Comment:
we're swallowing any errors that might happen from the API, since those will return with a non-200 status. Is that intentional?

Reply:
I carried over the previous method exactly as it was in fear of changing the behavior from how it was before.